src/addons/addonapi.cpp

```diff
@@ -92,6 +92,35 @@ void AddonApi::initialize() {
   }
 }
 
+void AddonApi::setTimedCallback(int interval, const QJSValue& callback) {
+  if (!callback.isCallable()) {
+    logger.debug() << "No callback received";
+    return;
+  }
+
+  // Creates a timer for `interval` msec before executing `callback`
+  QTimer* timer = new QTimer(this);
+  timer->setSingleShot(true);
+  timer->setInterval(interval);
+  m_timers.insert(m_addon->id(), timer);
+
+  connect((m_timers.value(m_addon->id())), &QTimer::timeout, this,
+          [callback]() { callback.call(); });
+
+  timer->start();
+}
+
+void AddonApi::clearTimedCallbacks() {
```

Comment:
Shouldn't the new name changes be made in the header file too. Should clearTimedCallbacks be named clearTimedCallback because only one callback is beeing cleared? Should setTimedCallback call clearTimedCallback to clear any previous callback?

Reply:
you beat me to it... working on removing the `clearTimedCallback` function altogether and just using a single timer object instead of a map. setting one timer will overwrite the other one